src/Workspaces/VisualBasic/Portable/Extensions/SyntaxTreeExtensions.vb

```diff
@@ -145,6 +146,23 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Extensions
             Return trivia.IsKind(SyntaxKind.SkippedTokensTrivia)
         End Function
 
+        <Extension()>
+        Public Function IsGlobalStatementContext(syntaxTree As SyntaxTree, position As Integer, cancellationToken As CancellationToken) As Boolean
+            If Not syntaxTree.IsScript() Then
+                Return False
+            End If
+
+            Dim token As SyntaxToken = syntaxTree.FindTokenOnLeftOfPosition(
+                position, cancellationToken, includeDirectives:=True).GetPreviousTokenIfTouchingWord(position)
+
+            If token.IsKind(SyntaxKind.None) Then
+                Dim compilationUnit = TryCast(syntaxTree.GetRoot(cancellationToken), CompilationUnitSyntax)
```

Comment:
This case occurs when the caret is at the beginning of submission/script or whilst typing the first word. It's necessary since there is no previous token to check against whether it was completed.
Case 1:
```
|
```
Case 2:
```
Dim|
```
Case 3:
```
|
Dim x = 12
```